src/Discord.Net.Commands/Attributes/Preconditions/RequireOwnerAttribute.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Discord;
+
+namespace Discord.Commands
+{
+    /// <summary>
+    /// Require that the command is invoked by the owner of the bot.
+    /// </summary>
+    /// <remarks>This precondition will only work if the bot is a bot account.</remarks>
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class RequireOwnerAttribute : PreconditionAttribute
+    {
+        private IApplication application;
+
+        public override async Task<PreconditionResult> CheckPermissions(CommandContext context, CommandInfo command, IDependencyMap map)
+        {
+            if (application == null)
+                application = await context.Client.GetApplicationInfoAsync();
```

Comment:
I didn't notice the caching here already... whoops. But yeah, never cache at the precondition level since it won't be updated by logouts.